models/sklearn/classification/classifierGRID.py

```diff
@@ -0,0 +1,153 @@
+                                          log_level=self.log_level).post_processing()
+
+            # define the length of parameters
+            parameters_grid = {'kernel': tr_process["kernel"],
+                               'C': tr_process["C"],
+                               'gamma': tr_process["gamma"],
+                               'class_weight': tr_process["balanceClasses"]
+                               }
+
+            # inner with K-Fold cross-validation declaration
+            random_seed = None
+            shuffle = self.config["k_fold_shuffle"]
+            if shuffle is True:
+                random_seed = self.config["seed"]
+            elif shuffle is False:
+                random_seed = None
+            self.logger.info("Fitting the data to the classifier with K-Fold cross-validation..")
+            inner_cv = KFold(n_splits=tr_process["n_fold"],
+                             shuffle=shuffle,
+                             random_state=random_seed
+                             )
+            # initiate GridSearch Object
+            gsvc = GridSearchCV(estimator=grid_clf,
+                                param_grid=parameters_grid,
+                                cv=inner_cv,
+                                n_jobs=self.config["parallel_jobs"],
+                                verbose=self.config["verbose"]
+                                )
+
+            self.logger.debug("Shape of X before train: {}".format(features_prepared.shape))
+            self.logger.info("Fitting the data to the model..")
+            gsvc.fit(features_prepared, self.y)
+
+            # print(gsvc.cv_results_["params"])
+            self.logger.info("Results from each best preprocess training:")
+            self.logger.info("a) Best score: {}".format(gsvc.best_score_))
+            self.logger.info("b) Best estimator: {}".format(gsvc.best_estimator_))
+            self.logger.info("c) Best parameters: {}".format(gsvc.best_params_))
+            self.logger.info("Counted evaluations in this GridSearch process: {}".format(len(gsvc.cv_results_["params"])))
+
+            # save best results for each train process
+            exports_dir = "{}_{}".format(self.config.get("exports_directory"), self.class_name)
+            results_path = FindCreateDirectory(self.exports_path,
+                                               os.path.join(exports_dir, "results")).inspect_directory()
+            results_best_dict_name = "result_{}_{}_best_{}.json"\
+                .format(self.class_name, tr_process["preprocess"], gsvc.best_score_)
+
+            results_dict = dict()
+            results_dict["score"] = gsvc.best_score_
+            results_dict["params"] = gsvc.best_params_
+            results_dict["n_fold"] = tr_process['n_fold']
+            results_dict["preprocessing"] = tr_process["preprocess"]
+            with open(os.path.join(results_path, results_best_dict_name), 'w') as grid_best_json:
```

Comment:
I'll have to look at this workflow in a bit more detail - it would be better to not have this method do too many things - e.g. it seems that it performs the training process and also saves this file. I would prefer that we split it up into more methods, for example one to do a specific training process, and then another one that loops through the settings, calls the training process, and then saves the results